orchestrator/modules/tools/execution/telemetry.py

```
@@ -152,10 +173,17 @@ async def write_telemetry(
         # Loud on purpose (PRD-185 S1): a DEBUG swallow here hid a 2-month,
         # 0-organic-rows telemetry outage. Failures must be visible.
         logger.warning(f"[telemetry] Failed to write tool execution log: {exc}")
-        try:
-            db.rollback()
-        except Exception:
-            pass
+        if db is not None:
+            try:
+                db.rollback()
+            except Exception:
+                pass
```

Ruff (0.16.0)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)